diff --git a/CMakeLists.txt b/CMakeLists.txt
index f83d986..87e3d03 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -438,6 +438,7 @@ if (WIN32)
         ${CUDA_CUBLAS_LIBRARIES}
         ${CUDA_cusparse_LIBRARY}
     )
+    target_compile_definitions( magma PRIVATE MAGMA_BUILDING_MAGMA )
 
     # no Fortran files at the moment (how to test libmagma_all_f is not empty?),
     # but keep this around for future reference.
diff --git a/include/magma_c.h b/include/magma_c.h
index c59a0a2..496dd8d 100644
--- a/include/magma_c.h
+++ b/include/magma_c.h
@@ -1774,8 +1774,8 @@ magma_cunmtr_m(
 // =============================================================================
 // MAGMA utility function definitions
 
-extern const magmaFloatComplex MAGMA_C_NAN;
-extern const magmaFloatComplex MAGMA_C_INF;
+MAGMA_DLL extern const magmaFloatComplex MAGMA_C_NAN;
+MAGMA_DLL extern const magmaFloatComplex MAGMA_C_INF;
 
 int magma_c_isnan( magmaFloatComplex x );
 int magma_c_isinf( magmaFloatComplex x );
diff --git a/include/magma_d.h b/include/magma_d.h
index 6ff826f..d827ed1 100644
--- a/include/magma_d.h
+++ b/include/magma_d.h
@@ -1774,8 +1774,8 @@ magma_dormtr_m(
 // =============================================================================
 // MAGMA utility function definitions
 
-extern const double MAGMA_D_NAN;
-extern const double MAGMA_D_INF;
+MAGMA_DLL extern const double MAGMA_D_NAN;
+MAGMA_DLL extern const double MAGMA_D_INF;
 
 int magma_d_isnan( double x );
 int magma_d_isinf( double x );
diff --git a/include/magma_s.h b/include/magma_s.h
index b970e6e..c9744ce 100644
--- a/include/magma_s.h
+++ b/include/magma_s.h
@@ -1774,8 +1774,8 @@ magma_sormtr_m(
 // =============================================================================
 // MAGMA utility function definitions
 
-extern const float MAGMA_S_NAN;
-extern const float MAGMA_S_INF;
+MAGMA_DLL extern const float MAGMA_S_NAN;
+MAGMA_DLL extern const float MAGMA_S_INF;
 
 int magma_s_isnan( float x );
 int magma_s_isinf( float x );
diff --git a/include/magma_types.h b/include/magma_types.h
index aab2929..c7c4699 100644
--- a/include/magma_types.h
+++ b/include/magma_types.h
@@ -37,6 +37,16 @@
   #endif
 #endif
 
+#ifdef _WIN32
+  #ifdef MAGMA_BUILDING_MAGMA
+    #define MAGMA_DLL __declspec(dllexport)
+  #else
+    #define MAGMA_DLL __declspec(dllimport)
+  #endif
+#else
+  #define MAGMA_DLL
+#endif
+
 
 // =============================================================================
 // To use int64_t, link with mkl_intel_ilp64 or similar (instead of mkl_intel_lp64).
diff --git a/include/magma_z.h b/include/magma_z.h
index 38fd189..f473012 100644
--- a/include/magma_z.h
+++ b/include/magma_z.h
@@ -1774,8 +1774,8 @@ magma_zunmtr_m(
 // =============================================================================
 // MAGMA utility function definitions
 
-extern const magmaDoubleComplex MAGMA_Z_NAN;
-extern const magmaDoubleComplex MAGMA_Z_INF;
+MAGMA_DLL extern const magmaDoubleComplex MAGMA_Z_NAN;
+MAGMA_DLL extern const magmaDoubleComplex MAGMA_Z_INF;
 
 int magma_z_isnan( magmaDoubleComplex x );
 int magma_z_isinf( magmaDoubleComplex x );
