diff --git a/sparse/control/magmasparse_internal.h b/sparse/control/magmasparse_internal.h
index 09799d0..3a23fe6 100644
--- a/sparse/control/magmasparse_internal.h
+++ b/sparse/control/magmasparse_internal.h
@@ -21,6 +21,10 @@ extern "C" {
 
 magma_int_t cusparse2magma_error( cusparseStatus_t status );
 
+#ifdef __cplusplus
+} // extern C
+#endif
+
 
 /**
     Macro checks the return code of a function;
@@ -74,8 +78,4 @@ magma_int_t cusparse2magma_error( cusparseStatus_t status );
     } while(0)
 
 
-#ifdef __cplusplus
-} // extern C
-#endif
-
 #endif        //  #ifndef MAGMASPARSE_INTERNAL_H
