From ecd302f2637d986f726938f4c1b5b64af05c5b0a Mon Sep 17 00:00:00 2001
From: Daniel Ching <carterbox@users.noreply.github.com>
Date: Wed, 26 Oct 2022 11:39:22 -0500
Subject: [PATCH 1/2] Add all target

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba53137..3b9edf1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -294,11 +294,19 @@ if (MAGMA_ENABLE_CUDA)
         message( STATUS "    compile for CUDA arch 8.0 (Ampere)" )
     endif()
 
+    if (GPU_TARGET MATCHES "all")
+      set( MIN_ARCH 370)
+      SET( NV_SM ${CUDA_ARCH_LIST})
+      separate_arguments(NV_SM)
+      SET( NV_COMP "")
+    endif()
+
     if (NOT MIN_ARCH)
         message( FATAL_ERROR "GPU_TARGET must contain one or more of Fermi, Kepler, Maxwell, Pascal, Volta, Turing, Ampere, or valid sm_[0-9][0-9]" )
     endif()
 
     set( CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} -Xcompiler -fPIC ${NV_SM} ${NV_COMP} ${FORTRAN_CONVENTION} )
+    MESSAGE(STATUS "CUDA_NVCC_FLAGS: ${CUDA_NVCC_FLAGS}")
     #add_definitions( "-DMAGMA_HAVE_CUDA -DMAGMA_CUDA_ARCH_MIN=${MIN_ARCH}" )
     set(MAGMA_HAVE_CUDA "1")
     set(MAGMA_CUDA_ARCH_MIN "${MIN_ARCH}")
-- 
2.24.0.windows.1

