diff --git a/sparse/control/magmasparse_internal.h b/sparse/control/magmasparse_internal.h
index 9600c9a..fb30fea 100644
--- a/sparse/control/magmasparse_internal.h
+++ b/sparse/control/magmasparse_internal.h
@@ -18,9 +18,11 @@
 extern "C" {
 #endif
 
+magma_int_t cusparse2magma_error(cusparseStatus_t status);
 
-magma_int_t cusparse2magma_error( cusparseStatus_t status );
-
+#ifdef __cplusplus
+} // extern C
+#endif
 
 /**
     Macro checks the return code of a function;
@@ -29,7 +31,7 @@ magma_int_t cusparse2magma_error( cusparseStatus_t status );
     Assumes variable info and label cleanup exist.
     Usually, all paths (successful and error) exit through the cleanup code.
     Example:
-    
+
         magma_int_t function()
         {
             magma_int_t info = 0;
@@ -42,40 +44,34 @@ magma_int_t cusparse2magma_error( cusparseStatus_t status );
             magma_free( B );
             return info;
         }
-    
+
     @see CHECK_CUSPARSE
     @ingroup magma_error_internal
     ********************************************************************/
-#define CHECK( err )             \
-    do {                         \
-        magma_int_t e_ = (err);  \
-        if ( e_ != 0 ) {         \
-            info = e_;           \
-            goto cleanup;        \
-        }                        \
-    } while(0)
-
+#define CHECK(err)                                                             \
+  do {                                                                         \
+    magma_int_t e_ = (err);                                                    \
+    if (e_ != 0) {                                                             \
+      info = e_;                                                               \
+      goto cleanup;                                                            \
+    }                                                                          \
+  } while (0)
 
 /**
     Macro checks the return code of a cusparse function;
     if non-zero, maps the cusparse error to a magma error and sets info,
     then does goto cleanup.
-    
+
     @see CHECK
     @ingroup magma_error_internal
     ********************************************************************/
-#define CHECK_CUSPARSE( err )                   \
-    do {                                        \
-        cusparseStatus_t e_ = (err);            \
-        if ( e_ != 0 ) {                        \
-            info = cusparse2magma_error( e_ );  \
-            goto cleanup;                       \
-        }                                       \
-    } while(0)
-
-
-#ifdef __cplusplus
-} // extern C
-#endif
+#define CHECK_CUSPARSE(err)                                                    \
+  do {                                                                         \
+    cusparseStatus_t e_ = (err);                                               \
+    if (e_ != 0) {                                                             \
+      info = cusparse2magma_error(e_);                                         \
+      goto cleanup;                                                            \
+    }                                                                          \
+  } while (0)
 
-#endif        //  #ifndef MAGMASPARSE_INTERNAL_H
+#endif //  #ifndef MAGMASPARSE_INTERNAL_H
